cli/jhipster.js

```diff
@@ -35,21 +34,23 @@ if (!semver.satisfies(currentNodeVersion, minimumNodeVersion)) {
     process.exit(1);
 }
 
-let preferGlobal = false;
-if (process.argv.length >= 3) {
-    if (process.argv[2] === 'upgrade') {
-        // Prefer global version for `jhipster upgrade` to get most recent code
-        preferGlobal = true;
-    }
+let preferLocal = true;
+
+// Don't use commander for parsing command line to avoid polluting it in cli.js 
+// --prefer-local: Always resolve node modules locally (useful when using linked module)
+if ((process.argv.indexOf('upgrade') > -1) && (process.argv.indexOf('--prefer-local') < 0)) {
```

Comment:
you could also do `process.argv.includes('upgrade') && process.argv.includes('--prefer-local'))` which seems cleaner.